test/execution/test_protocol.py

```diff
@@ -2,16 +2,33 @@
 import torch as th
 
 import syft as sy
+from syft.execution.role import Role
 
 
-def test_trace_communication_actions(workers):
-    bob = workers["bob"]
+def test_func2protocol_creates_roles():
+    @sy.func2protocol(roles=["alice", "bob"])
+    def protocol(roles):
+        alice = roles["alice"]
+
+        tensor = alice.fetch(th.tensor([1]))
+
+        return tensor
+
+    assert protocol.is_built
+    assert len(protocol.roles) == 2
+    assert isinstance(protocol.roles["alice"], Role)
+    assert isinstance(protocol.roles["bob"], Role)
+
 
-    @sy.func2protocol(args_shape={"alice": ((1,),)})
+def test_trace_communication_actions_send():
+    @sy.func2protocol(roles=["alice", "bob"], args_shape={"alice": ((1,),)})
     def protocol(roles):
-        tensor = roles["alice"].fetch(th.tensor([1]))
+        alice = roles["alice"]
+        bob = roles["bob"]
 
-        tensor.send(bob)
+        tensor = alice.fetch(th.tensor([1]))
+
+        tensor.send(bob.worker)
```

Comment:
I don't love this syntax and I'm not sure it will work when the `Roles` are executed, but it does make the tracing tests pass. 🤷

Reply:
Let's polish that later?